### PR TITLE
[DOP-19901] - add integration MSSQL tests & CI

### PR DESCRIPTION
**Pull request by maxim-lixakov**
## Change Summary

- add integration `MSSQL` tests
- add github actions CI to run tests

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/worker/handlers/db/mssql.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    transfer_dto: MSSQLTransferDTO
+
+    def connect(self, spark: SparkSession):
+        self.connection = MSSQL(
+            host=self.connection_dto.host,
+            port=self.connection_dto.port,
+            user=self.connection_dto.user,
+            password=self.connection_dto.password,
+            database=self.connection_dto.database_name,
+            extra=self.connection_dto.additional_params or {"trustServerCertificate": "true"},
```
Comment (dolfinus) [inline syncmaster/worker/handlers/db/mssql.py:31]: What about merging `{"trustServerCertificate": "true"}` with additional_params?

File: tests/test_integration/test_run_transfer/test_mssql.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    assert "password" not in target_auth_data
+    reader = DBReader(
+        connection=mssql,
+        table="dbo.target_table",
+    )
+    df = reader.run()
+
+    # as spark rounds datetime up to milliseconds while writing to mssql: https://onetl.readthedocs.io/en/latest/connection/db_connection/mssql/types.html#id5
+    df = df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
+    init_df = init_df.withColumn("REGISTERED_AT", date_trunc("second", col("REGISTERED_AT")))
```
Comment (dolfinus) [inline tests/test_integration/test_run_transfer/test_mssql.py:123]: Is that really needed?

File: tests/test_integration/test_run_transfer/test_mssql.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+    for field in init_df.schema:
+        df = df.withColumn(field.name, df[field.name].cast(field.dataType))
+
+    assert df.sort("ID").collect() == init_df.sort("ID").collect()
```
Comment (dolfinus) [inline tests/test_integration/test_run_transfer/test_mssql.py:128]: Which case MSSQL uses by default, UPPERCASE or lowercase?

File: syncmaster/worker/handlers/db/mssql.py (hunk 1 above)
Comment (dolfinus) [reply syncmaster/worker/handlers/db/mssql.py]: No, vice versa

File: tests/test_integration/test_run_transfer/test_mssql.py (hunk 3 above)
Comment (IlyasDevelopment) [reply tests/test_integration/test_run_transfer/test_mssql.py:128]: Without column names normalization MSSQL uses lowercase

Comment (dolfinus) [reply tests/test_integration/test_run_transfer/test_mssql.py:128]: In this case, shouldn't we compare like this?
```suggestion
    assert df.sort("id").collect() == init_df.sort("id").collect()
```

File: tests/test_integration/test_run_transfer/test_mssql.py (hunk 2 above)
Comment (IlyasDevelopment) [reply tests/test_integration/test_run_transfer/test_mssql.py:123]: In MSSQL the datetime type [rounds values](https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-ver16#:~:text=8%20bytes-,Accuracy,-Rounded%20to%20increments) to the nearest increment of 0.000, 0.003, or 0.007 seconds (approx. 3.33 milliseconds) due to its precision limitation. For example:

- `id=7282, 2006-02-14 12:36:15.589307` is rounded to 590 milliseconds
- `id=815, 1998-11-10 01:46:12.228133` is rounded to 227 milliseconds

Therefore I decided to drop milliseconds in tests

Comment (dolfinus) [reply tests/test_integration/test_run_transfer/test_mssql.py:123]: I see, another candidate to fix in spark-dialect-extension

**Review by dolfinus — APPROVED** [review]

**Pull request merged**